pyproject.toml

```diff
@@ -7,9 +7,13 @@ readme = "README.md"
 
 [tool.poetry.dependencies]
 python = ">=3.8,<3.13"
+numpy = [
+   {version = "<=1.24.*", python = "<3.9"},
+   {version = ">=1.25.*", python = ">=3.9"}
+]
 
 [tool.poetry.extras]
-dev = ["pytest", "black", "isort", "coverage", "pytest-cov", "sphinx", "sphinx-autodoc-typehints"]
+dev = ["pytest", "black", "isort", "coverage", "pytest-cov", "sphinx", "sphinx-autodoc-typehints","matplotlib"]
```

Comment:
one of your core functions is using matplotlib
so why is it in the dev dependencies?